### PR TITLE
#24 feat: add login register page design

### LINKED ISSUE #11: [P0] User login, register page

### PR DESCRIPTION
**Pull request by ohsung1125**
This is the inital design of the Login + Register Page.
It made it using the header on the main page.

1) Login Page
It is composed that Input components [ Email and Password ] and Button Login and Goto-Register-Page.
This Page is chaning Components-color according to Light-Dark Theme.

2) Register Page
It is composed that Input components [ Email, Password, CheckPassword,  Name, PhoneNumber, Certified ], Certified-Button [ Email, PhoneNumber, Certified-Number ] and Complete-Regist Button.
This Page is chaning Components-color for the same reason on LoginPage.

Please let me know if you need to modify the design or if there is a bug!

Closes #11.

### CODE REVIEW

**Comment by seiwonpark** [comment]
**Remember**:
- Add reviewers (@sanghyun1128 @SeiwonPark)
- Assign yourself
- Add labels

I've done it for you but from the next PR, please check these.

File: web/src/routes/LoginPage.tsx [hunk 1]
```diff
@@ -1,5 +1,127 @@
...
+
+import BackgroundImage from '../assets/main-page.webp';
+import Header from '../components/Header';
+import { useTheme } from '../hooks/useTheme';
+
+// 로그인 페이지 모르는 점
+// 1. 가로 짝대기 어떻게 생성하는지
+// 2. 로그인 제목이 더 굵게 안됨.
+// 3. 이메일과 비밀번호 font-color가 바뀌지 않음.
+// 4. 로그인컨테이너의 최소폭을 설정하고 싶은데 모름.
```
Comment (seiwonpark) [inline web/src/routes/LoginPage.tsx]: Please remove unrecognizable comments.

File: web/src/routes/LoginPage.tsx [hunk 2]
```diff
@@ -1,5 +1,127 @@
...
+  min-width : 20vh;
+  height: 50vh;
+  background-color: ${(props) => props.theme.colors.container};
+  box-shadow: 0 10px 15px rgba(0, 0, 0, 0.3);
+`;
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+
```
Comment (seiwonpark) [inline web/src/routes/LoginPage.tsx]: Please remove all whitespaces and blank spaces. Or please run

```bash
$ npm run lint:fix && npm run format
```

FYI. In your local Visual Studio Code, install extensions for **Prettier**, **ESLint** and check `Format on Save` to apply auto save and format.

File: web/src/routes/LoginPage.tsx [hunk 3]
```diff
@@ -1,5 +1,127 @@
...
+  }
+  :-ms-input-placeholder {
+    opacity: 0.4;
+  }
+`;
+
+const LoginTitle = styled.span`
+  text-align: left;
+  font-size: 5vh;
+  font-weight: 1000;
```
Comment (seiwonpark) [inline web/src/routes/LoginPage.tsx]: Q. Does our font(`Inter`, `san-serif`) support font-weight `1000`? See https://fonts.google.com/specimen/Inter?query=inter its maximum is `900`.

File: web/src/routes/RegisterPage.tsx [hunk 4]
```diff
@@ -1,5 +1,188 @@
...
+
+const FontText = styled.span`
+  margin-top: 5px;
+  text-align: left;
+  font-size: 1.3vh;
+  font-weight: 600;
+  color: ${(props) => props.theme.colors.default};
+`;
+
+const PhoneNumberBox = styled.button`
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: You sure this is a `<button>`? Though we allow E.164 format, `<select>` seems better. Or, if we only allow korean phone numbers, this doesn't have to be a `<button>` but just `<div>`.

File: web/src/routes/RegisterPage.tsx [hunk 5]
```diff
@@ -1,5 +1,188 @@
...
+  border: 3px solid #ebeff7;
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: We have a global `border` value here: https://github.com/kookmin-sw/capstone-2023-40/blob/feature/login-regist/web/src/types/styled.d.ts#L8 

Please make use of reusable values not hard coded one.

File: web/src/routes/RegisterPage.tsx [hunk 6]
```diff
@@ -1,5 +1,188 @@
...
+  padding: 1.7vh;
+  font-size: 1.5vh;
+  font-weight: 700;
+  margin-right: 1vh;
+  color: ${(props) => props.theme.colors.text};
+  background-color: white;
+  border-radius: ${(props) => props.theme.borderRadius};
+`;
+
+const Button = styled.button`
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: https://github.com/kookmin-sw/capstone-2023-40/pull/24/files#diff-0c351720713fe6bc89fef12c50d65c5bd4467bf7b7f75f099bf20e5cf381b639R115 This component has its name `CertifyButton` but this button has no name. Please name it!

File: web/src/routes/RegisterPage.tsx [hunk 7]
```diff
@@ -1,5 +1,188 @@
...
-  return <div>RegisterPage</div>;
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+import BackgroundImage from '../assets/main-page.webp';
+import Header from '../components/Header';
+import { useTheme } from '../hooks/useTheme';
+
+// 핸드폰 번호 + 인증코드 제약 걸기
+// 이메일 regex
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: Please remove unrecognizable comments.

File: web/src/routes/LoginPage.tsx [hunk 8]
```diff
@@ -1,5 +1,127 @@
...
+
+const Input = styled.input`
+  padding: 1.7vh;
+  margin-top: 10px;
+  margin-bottom: 10px;
+  border: 3px solid #ebeff7;
+  border-radius: 16px;
+  font-size: 1.5vh;
+  font-weight: 600;
+  flex: 1;
```
Comment (seiwonpark) [inline web/src/routes/LoginPage.tsx:44]: Please add a blank line here to separate events

```
flex: 1;
                        <------- add blank line to divide events
&:focus {
...
```

File: web/src/routes/RegisterPage.tsx [hunk 9]
```diff
@@ -1,5 +1,188 @@
...
+`;
+
+const RegistContainer = styled.div`
+padding: 7vw;
+margin-left: calc(20vw - 5vmin);
+margin-right: calc(20vw - 5vmin);
+min-width : 30vh;
+height: 80vh;
+background-color: ${(props) => props.theme.colors.container};
+box-shadow: 0 10px 15px rgba(0, 0, 0, 0.3);
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: Please add indentation here.

File: web/src/routes/RegisterPage.tsx [hunk 10]
```diff
@@ -1,5 +1,188 @@
...
+            <CertifyButton onClick={() => alert("클릭됨")} type="submit" theme={theme}>
+              인증요청
+            </CertifyButton>
+          </ContainerBox>
+
+          <FontText theme={theme}>비밀번호</FontText>
+          <Input type="password" theme={theme} placeholder="비밀번호를 입력해주세요." />
+
+          <FontText theme={theme}>비밀번호 확인</FontText>
+          <Input type="password" theme={theme} placeholder="비밀번호를 한번더 입력해주세요." />
```
Comment (seiwonpark) [inline web/src/routes/RegisterPage.tsx]: 한번더(x) -> 한 번 더(o)

**Review by seiwonpark — CHANGES_REQUESTED** [review]
Please update your code on your `feature/login-regist` branch, and after make changes, push to pull request again.

File: web/src/routes/LoginPage.tsx (hunk 2 above)
Comment (ohsung1125) [reply web/src/routes/LoginPage.tsx]: Install extensions Prettier and Eslint, and npm run lint:fix. but it was error [ No files matching the pattern "'./src/**/*.{ts,tsx,js,jsx}'" were found. Please check for typing mistakes in the pattern. ]

Comment (seiwonpark) [reply web/src/routes/LoginPage.tsx]: Well let me figure it out

**Review by seiwonpark — APPROVED** [review]
LGTM. Thanks!

File: web/src/routes/LoginPage.tsx (hunk 2 above)
Comment (seiwonpark) [reply web/src/routes/LoginPage.tsx]: After 1:1 w/ @ohsung1125 ,

resolved as follows:

```
"lint": "eslint \"./src/**/*.{ts,tsx,js,jsx}\"",
"lint:fix": "eslint --fix \"./src/**/*.{ts,tsx,js,jsx}\"",
"format": "prettier --write --config ./.prettierrc \"./src/**/*.{ts,tsx,js,jsx}\""
```

**Review by sanghyun1128 — APPROVED** [review]
LGTM!

**Pull request merged**